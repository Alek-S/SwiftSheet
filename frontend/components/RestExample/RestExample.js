import React from 'react';
import PropTypes from 'prop-types';
import { ApiCard } from '../../defaultStyle';

/**
 * @function
 * Card providing link to rest api endpoint of sheet
 * @param {string} - props.host - host origin
 * @param {string} - props.id - sheet ID
 * @returns {jsx} <RestExample />
 */
const RestExample = ({ host, id }) => (
	<ApiCard>
		<span className="pill">REST API:</span>
		<a href={`${host}/api/sheet/${id}`} target="_blank">
			{host}/api/sheet/{id}
		</a>
	</ApiCard>
);

RestExample.propTypes = {
	host: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
};

export { RestExample };
