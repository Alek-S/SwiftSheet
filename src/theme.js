/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
