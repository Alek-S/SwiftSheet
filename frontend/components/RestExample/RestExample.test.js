import React from 'react';
import 'jest-styled-components';

import { RestExample } from './RestExample';
import theme from '../../theme';

describe('RestExample', () => {
	test('matches the snapshot', () => {
		const tree = shallow(
			<RestExample host="mockHost.com" id="42" theme={theme} />
		);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
