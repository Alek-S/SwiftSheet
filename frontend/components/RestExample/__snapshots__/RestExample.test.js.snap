// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestExample matches the snapshot 1`] = `
.c0 {
  width: 500px;
  padding: 1rem;
  margin: auto;
  margin-top: 1rem;
  background-color: #424242;
  color: #BCBCBC;
}

.c0 a {
  margin-left: 1rem;
  color: #68F6BA;
}

.c0 .pill {
  background: linear-gradient(160deg,#57DACB 0%,#55A2ED 100%);
  color: white;
  padding: 0.1rem 0.5rem;
  border-radius: 4px;
}

.c0 .pill-secondary {
  background-color: #595959;
  color: white;
  padding: 0.1rem 0.5rem;
  border-radius: 4px;
}

.c0 .gql_query {
  margin-left: 0.5rem;
  color: #EAEAEA;
}

.c0 .query_title {
  margin-top: 0.75rem;
}

<defaultStyle__Card
  className="c0"
>
  <span
    className="pill"
  >
    REST API:
  </span>
  <a
    href="mockHost.com/api/sheet/42"
    target="_blank"
  >
    mockHost.com
    /api/sheet/
    42
  </a>
</defaultStyle__Card>
`;
